blocks/subscriptions-block/components/PaywallOfferBody.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react';
+import isUrl from 'is-url';
+
+import { useIdentity } from '@wpmedia/identity-block';
+import { useOffer } from './Offer';
+import { usePaywall } from './Paywall';
+
+export const PaywallOfferBody = ({
+  loginURL,
+  offerURL = '/offer/',
+  campaignCode
+}) => {
+  const { isLoggedIn } = useIdentity();
+  const [title, setTitle] = useState();
+  const [subtitle, setSubtitle] = useState();
+
+  const {
+    results: paywallResults = {},
+    ArcP = {}
+  } = usePaywall();
+  const rules = ArcP._rules || [];
+  const [ triggeredRule, setTriggeredRule ] = useState();
+  const [ triggeredRules, setTriggeredRules ] = useState([]);
+  const [ pw, setPw ] = useState();
+
+  const { offer, fetchOffer } = useOffer({campaignCode: campaignCode ? campaignCode : !isUrl(pw) ? pw : null});
+  const [ pwOffer, setPwOffer ] = useState();
+  useEffect(() => {
+    const selectedOffer = pwOffer ? pwOffer : offer;
+    console.log('fffers', pwOffer, offer);
+    if (selectedOffer) {
+      setTitle(selectedOffer.pageTitle);
+      setSubtitle(selectedOffer.pageSubTitle);
+      console.log('setting selectedOffer title', selectedOffer);
+    }
+  }, [ offer, pwOffer ]);
+
+  useEffect(() => {
+    const fetchNewOffer = async () => {
+      setPwOffer(await fetchOffer(pw));
+    };
+    if (pw && !isUrl(pw)) {
+      fetchNewOffer();
+    }
+  }, [pw]);
+
+  useEffect(() => {
+    const ruleId = paywallResults.triggered && paywallResults.triggered.id;
+    const rc = paywallResults.triggered && paywallResults.triggered.rc;
+    if (ruleId) {
+      const triggeredRule = rules.find(rule => ruleId === rule.id);
+      const triggeredRules = rules.filter(rule => rule.rt[1] < rc);
+      setTriggeredRule(triggeredRule);
+      setTriggeredRules(triggeredRules);
+    }
+  }, [paywallResults.triggered, rules ]);
```

Comment:
same with this and `triggeredRules`. is this something that should be in the dependencies